libs/core/src/form/Field.ts

```diff
@@ -189,19 +160,57 @@ export class Field<Type = unknown, Props extends ComponentProps = ComponentProps
   };
 
   public setValue(value: Type): void {
-    this.ref.value = value;
+    if (isRef(this.ref)) {
+      this.ref.value = value;
+    } else {
+      // @ts-expect-error todo
+      this.ref = value;
+    }
   }
 
   public setDisabled(value: boolean): void {
-    this.isDisabled.value = value;
+    if (isRef(this.isDisabled)) {
+      this.isDisabled.value = value;
+    } else {
+      // @ts-expect-error update types to for when the refs are unwrapped
+      this.isDisabled = value;
+    }
   }
 
   public setOptional(value: boolean): void {
-    this.isOptional.value = value;
+    if (isRef(this.isOptional)) {
+      this.isOptional.value = value;
+    } else {
+      // @ts-expect-error update types to for when the refs are unwrapped
+      this.isOptional = value;
+    }
   }
 
   public setReadOnly(value: boolean): void {
-    this.isReadOnly.value = value;
+    if (isRef(this.isReadOnly)) {
+      this.isReadOnly.value = value;
+    } else {
+      // @ts-expect-error update types to for when the refs are unwrapped
+      this.isReadOnly = value;
+    }
+  }
+
+  public addError(error: string): void {
+    if (isRef(this.errors)) {
+      this.errors.value.push(error);
+    } else {
+      // @ts-expect-error update types to for when the refs are unwrapped
+      this.errors.push(error);
+    }
+  }
+
+  public setInvalid(): void {
+    if (isRef(this.isValid)) {
```

Comment:
maak hier even een helper func van dan, als die er nog niet al is

Reply:
toegevoegd